### PR TITLE
add samurai package

### PR DESCRIPTION
**Pull request by gouarin**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by gouarin** [comment]
@conda-forge/help-c-cpp

File: recipes/samurai/bld.bat [hunk 1]
```diff
@@ -0,0 +1,5 @@
+cmake ${CMAKE_ARGS}  -G "Ninja" -D CMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% %SRC_DIR% -B build -D CMAKE_SYSTEM_PROCESSOR=x64
```
Comment (wolfv) [inline recipes/samurai/bld.bat:1]: Does the `${CMAKE_ARGS}` syntax really work here? I would think that only `%CMAKE_ARGS%` works in `cmd.exe`.

File: recipes/samurai/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  url: https://github.com/hpc-maths/samurai/archive/v{{ version }}.tar.gz
+  sha256: e0459f3e881b64d2e4bd99dcc5d2167265620616bf5a6e51db192f321a1a3b56
+  # sha256 is the preferred checksum -- you can get it for a file with:
+  #  `openssl sha256 <file name>`.
+  # You may need the openssl package, available on conda-forge:
+  #  `conda install openssl -c conda-forge``
+
+build:
+  number: 0
+  skip: true  # [win and vc<14]
```
Comment (wolfv) [inline recipes/samurai/meta.yaml:18]: you can remove the `win and vc<14` as that version isn't in use anymore.

File: recipes/samurai/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+about:
+  home: https://github.com/hpc-maths/samurai
+  summary: 'Intervals coupled with algebra of set to handle adaptive mesh refinement and operators on it'
+  description: |
+    samurai allows to implement several mesh adaptation methods (patch-based, cell-based, point-based)
+    from the same data structure. The mesh is represented as intervals and a set algebra allows to efficiently
+    search for subsets among these intervals. Samurai also offers a flexible and pleasant interface to easily
+    implement numerical methods.
+  license: BSD-3-Clause
+  license_family: BSD
```
Comment (wolfv) [inline recipes/samurai/meta.yaml:60]: you can remove license_family since we use SPDX identifiers now and that's enough

**Pull request merged**